### DISCUSSION TITLE
Can't select ollama models

### DISCUSSION THREAD

**Discussion by eliezersouzareis** [open-webui/open-webui#1565] [General]
# Bug Report

## Description

**Bug Summary:**
The dropdown to select models in the application is not functioning as expected.

**Steps to Reproduce:**
1. Access the application.
2. Navigate to the dropdown to select models.
3. Attempt to select a model.

**Expected Behavior:**
When selecting a model from the dropdown, it should activate or display relevant information.

**Actual Behavior:**
Selecting a model from the dropdown does not trigger any action or display relevant information.

## Environment

- **Operating System:** Manjaro
- **Browser (if applicable):** Firefox 124.0.1

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
here is the most relevant logs
```
open-webui  | Traceback (most recent call last):
open-webui  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 411, in run_asgi
open-webui  |     result = await app(  # type: ignore[func-returns-value]
open-webui  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
open-webui  |     return await self.app(scope, receive, send)
open-webui  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
open-webui  |     await super().__call__(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
open-webui  |     await self.middleware_stack(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
open-webui  |     await self.app(scope, receive, _send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__
open-webui  |     with collapse_excgroups():
open-webui  |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
open-webui  |     self.gen.throw(typ, value, traceback)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/responses.py", line 261, in wrap
open-webui  |     await func()
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 217, in stream_response
open-webui  |     return await super().stream_response(send)
open-webui  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/responses.py", line 250, in stream_response
open-webui  |     async for chunk in self.body_iterator:
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 181, in body_stream
open-webui  |     raise app_exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 151, in coro
open-webui  |     await self.app(scope, receive_or_disconnect, send_no_error)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 85, in __call__
open-webui  |     await self.app(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__
open-webui  |     with collapse_excgroups():
open-webui  |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
open-webui  |     self.gen.throw(typ, value, traceback)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/responses.py", line 261, in wrap
open-webui  |     await func()
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 217, in stream_response
open-webui  |     return await super().stream_response(send)
open-webui  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/responses.py", line 250, in stream_response
open-webui  |     async for chunk in self.body_iterator:
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 181, in body_stream
open-webui  |     raise app_exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 151, in coro
open-webui  |     await self.app(scope, receive_or_disconnect, send_no_error)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
open-webui  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
open-webui  |     await app(scope, receive, sender)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
open-webui  |     await self.middleware_stack(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
open-webui  |     await route.handle(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 485, in handle
open-webui  |     await self.app(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
open-webui  |     await super().__call__(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
open-webui  |     await self.middleware_stack(scope, receive, send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
open-webui  |     await self.app(scope, receive, _send)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__
open-webui  |     with collapse_excgroups():
open-webui  |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
open-webui  |     self.gen.throw(typ, value, traceback)
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
open-webui  |     raise exc
open-webui  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__
open-webui  |     response = await self.dispatch_func(request, call_next)
open-webui  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/app/backend/apps/ollama/main.py", line 76, in check_url
open-webui  |     await get_all_models()
open-webui  |   File "/app/backend/apps/ollama/main.py", line 158, in get_all_models
open-webui  |     app.state.MODELS = {model["model"]: model for model in models["models"]}
open-webui  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
open-webui  |   File "/app/backend/apps/ollama/main.py", line 158, in <dictcomp>
open-webui  |     app.state.MODELS = {model["model"]: model for model in models["models"]}
open-webui  |                         ~~~~~^^^^^^^^^
open-webui  | KeyError: 'model'
open-webui  | INFO:apps.openai.main:get_all_models()
```

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Installed ollama using package manager then set up ollama as service and to listen on port 11434.
After that I cobbled the following docker compose file
```yml
version: '3.8'

services:
  open-webui:
    container_name: open-webui
    image: ghcr.io/open-webui/open-webui:main
    network_mode: 'host'
    ports:
      - "3000:8080"
    extra_hosts:
      - 'host.docker.internal:host-gateway'
    volumes:
      - open-webui:/app/backend/data
    environment:
      - OLLAMA_BASE_URL=http://127.0.0.1:11434
    restart: always

volumes:
  open-webui:
```
and ran `docker compose up`

## Additional Information

- The application is running on Docker (image: ghcr.io/open-webui/open-webui:main).
- Ollama version: 0.1.17 is installed externally on my computer.
- It seems to be a development error since the API documentation for `/api/tags` (https://github.com/ollama/ollama/blob/v0.1.32/docs/api.md#list-local-models) shows that there is no 'model' key.
- Even after fixing the current issue (by changing `model["model"]` to `model["name"]`), the list of models shows up but cannot be selected (clicking them does not trigger any action).

**Comment by ghost** [comment]
>     * Ollama version: 0.1.17 is installed externally on my computer.

That is an ancient version of Ollama is there a good reason you're using it? I'm fairly certain that there are [vulnerabilities like this](https://github.com/advisories/GHSA-5jx5-hqx5-2vrj). I'm not sure that we're compatible with it anymore.  See this commit which deprecates `name` from many types and adds `model` everywhere: ollama/ollama@1f5b7ff976460ad5305ab31998554944825f1f5f

**Reply by eliezersouzareis** [reply]
> That is an ancient version of Ollama is there a good reason you're using it ? 

I installed it and forgot about it. Then I mistakenly recalled it being installed from the package manager. It was pretty late (3 AM), and in my sleep-deprived state, I didn't even think about checking the version in the package manager.

Many such cases 🥲.

Thank you for taking the time to answer, and I apologize for the non-issue.

On a side note, could the README.md explicitly state which version of Ollama Open WebUI is compatible with?

**Reply by justinh-rahb** [reply]
> On a side note, could the README.md explicitly state which version of Ollama Open WebUI is compatible with?

I believe our docs do say that one should ensure they are running the latest versions of everything for best compatibility.

**Reply by ghost** [reply]
> Thank you for taking the time to answer, and I apologize for the non-issue.

@eliezersouzareis 🥂 😀 

> On a side note, could the README.md explicitly state which version of Ollama Open WebUI is compatible with?

@justinh-rahb You're correct! It's mentioned multiple times throughout including the [General Connection Errors troubleshooting document](https://docs.openwebui.com/getting-started/troubleshooting/#general-connection-errors).

It's important to remember that LLM technology in general is under rapid development (I sound like a bot LoL).  Perhaps tracking specific compatibility versions might make sense in the future but until development and APIs mature and stabilize, I don't think it's warranted.

**Reply by eliezersouzareis** [reply]
Uhhhh... Perhaps I haven't been clear enough about what I meant and thought when using the word explicit. "Latest version" means nothing when you're quickly reading through documentation. Especially in the era of generative AIs where the information that matters is increasingly scattered in filler text. A table with compatibility information would be objectively better because firstly, it's not a word within a paragraph, and secondly, it's a completely different graphical element. That provides visibility.

For example, a table like this:

| Version | Ollama |
|---|---|
| v0.1.118(Current) | >= v0.1.32 |
| v0.1.117 | > v0.0.21 && < v0.1.32 |

carries much more information and is much more visible than writing "check that you have the latest version".

Obviously, this is just a suggestion, especially (as @lainedfles said) considering that neither open webui nor ollama have reached version 1.0.0.

On the other hand, personally, I think ollama will never release a version 1.0.0, and will follow in the footsteps of react-native (lol, lmao even), because the major version 0 guarantees unlimited powers (["Anything MAY change at any time"](https://semver.org/#spec-item-4)) and that's a very good protection a project can have when it comes to open-source. So waiting until the software matures to track compatibility is really a moot point, in my view.

Anyway, I started to digress, so... 🤷‍♂️

**Reply by justinh-rahb** [reply]
@eliezersouzareis I think we're going to have to agree to disagree a bit here.. maintaining a comprehensive compatibility matrix table does not sound like very good use of dev time testing different (especially historical) versions with each other. By all means if you'd like to take that on and submit a docs PR, nobody would say no to it I think.

For our purposes I think it's much simpler to say "Ensure you have pulled/installed the latest version of Ollama when using the latest version of WebUI". We would tell anyone posting an issue with an outdated version of either to update first anyway.

**Reply by eliezersouzareis** [reply]
@justinh-rahb What? no. That is not what I am suggesting. You do not even need to do a comprehensive matrix table, just list the current working version number for `ollama` when you release a new version, and then format it as a table for better visibility. That is not a great deal of work.    
I don't think this discussion will bring any new insight on these matters, so I will be closing it.    
Regardless, thank you for your work and contribution on this project, I appreciate it.  🥂 Cheers!
